app/controllers/admin/schedules_controller.rb

```diff
@@ -31,14 +31,17 @@ def update
         error_message = "Could not find event GUID: #{params[:event]}"
       end
 
+      event_schedule = event.event_schedules.find_by(schedule_id: params[:schedule])
+
       if params[:date] == 'none'
-        event.start_time = nil
-        event.room = nil
-        event.save!
+        event_schedule.destroy if event_schedule.present?
         render json: { 'status' => 'ok' }
         return
       end
+
+      event_schedule = EventSchedule.create(event: event, schedule_id: params[:schedule]) unless event_schedule.present?
```

Comment:
But now `room` and `start_time` attributes are in `EventSchedule `and not in `Event`, so if the `event_schedule` is not created for this `schedule` I can not assign the attributes, so I think I have to create it here, when it is scheduled.